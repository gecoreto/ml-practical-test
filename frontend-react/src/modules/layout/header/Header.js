import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import searchImg from '../../../assets/img/search.svg';
import { cleanItems } from '../../items/store/itemsSlice';
import './header.sass';

export const Header = () => {
    const [search, setSearch] = useState('');
    const history = useHistory();
    const dispatch = useDispatch();
    const goToHome = () => {
        dispatch(cleanItems())
        history.push(`/`)
      };
    const handleSubmit = (e) => {
        let queryString = "search=" + search;
        history.replace(`/items?${queryString}`);
        history.go()
        e.preventDefault();
    }
    return (
        <header
            role="banner"
            className="ml-header-container py-1">
            <nav className="navbar container">
                <a id="nav-search-brand" className="navbar-brand mr-2" onClick={() => goToHome()} />
                <div className="navbar-search">
                    <form id="nav-search-form" onSubmit={(e) => handleSubmit(e)}>
                        <input
                            id="nav-search-input"
                            className="nav-search-input"
                            placeholder="Nunca dejes de buscar"
                            type="text"
                            value={search}
                            onChange={(e) => setSearch(e.target.value)}
                            aria-label="Search" />
                        <button
                            id="nav-search-button"
                            className="nav-search-btn"
                            type="submit">
                            <img
                                src={searchImg}
                                alt="buscar" />
                        </button>
                    </form>
                </div>
            </nav>
        </header >
    );
}

export default Header