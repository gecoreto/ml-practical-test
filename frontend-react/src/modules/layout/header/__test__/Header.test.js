import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

const mockHistoryReplace = jest.fn();
const mockHistoryGo = jest.fn();
const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => {
    return {
        useHistory: () => ({
            push: mockHistoryPush,
            replace: mockHistoryReplace,
            go: mockHistoryGo,
        }),
    };
});
jest.mock("react-redux", () => {
    return {
        useDispatch: () => jest.fn()
    };
});

describe("Header component", () => {

    const wrapper = shallow(<Header />);

    test('should test Header component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe("WHEN the Button is pressed", () => {

        it("THEN getProductList should be called", () => {
            const event = {
                preventDefault() { },
                target: { value: 'the-value' }
            };
            wrapper.find(`[id="nav-search-input"]`).props().onChange(event)
            expect(wrapper.find(`[id="nav-search-input"]`).prop('value')).toEqual('the-value')
            wrapper.find(`[id="nav-search-form"]`).props().onSubmit(event)
            expect(mockHistoryReplace).toHaveBeenCalledWith('/items?search=the-value')
        });
    });

    describe("WHEN the Image is pressed", () => {

        it("THEN push location should be called /", () => {
            wrapper.find(`[id="nav-search-brand"]`).simulate("click");
            expect(mockHistoryPush).toHaveBeenCalledWith('/');
        });
    });
});
