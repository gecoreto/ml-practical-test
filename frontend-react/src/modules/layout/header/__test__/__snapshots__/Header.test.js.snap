// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should test Header component 1`] = `ShallowWrapper {}`;
